client/src/context/AuthContext.tsx

```diff
@@ -0,0 +1,76 @@
+import { ReactNode, useCallback, useEffect, useMemo, useState } from "react";
+import { setAuthToken } from "../services/api.service";
+import {
+  AuthContext,
+  AuthContextValue,
+  TOKEN_STORAGE_KEY,
+  USER_STORAGE_KEY,
+} from "./auth-context";
+
+type AuthProviderProps = {
+  children: ReactNode;
+};
+
+export function AuthProvider({ children }: AuthProviderProps) {
+  const [token, setToken] = useState<string | null>(() => {
+    if (typeof window === "undefined") {
+      return null;
+    }
+    return window.localStorage.getItem(TOKEN_STORAGE_KEY);
```

Comment:
The token won’t be stored in local storage. Instead, it will be sent to the client as a secure, HTTP-only cookie (inaccessible to JavaScript). The server will hydrate the session by attaching it to window on render, and it will do so again on each reload

Reply:
Something like, window.userSession -- which will contain the user if they are logged in. Then that can be stored in context or redux